### PR TITLE
refactor: #160 moved page title to header

### PR DESCRIPTION
**Pull request by MaratReactJs**

### CODE REVIEW

File: mw-webapp/src/component/header/Header.tsx [hunk 1]
```diff
@@ -25,14 +25,16 @@ export const Header = () => {
...
         {LOGO_TEXT.toUpperCase()}
       </h1>
       {user && <h2 className={styles.title}>
-        Hello,
+        Welcome
         {" "}
         {user.displayName}
-        !
+        {" "}
+        {" "}
```
Comment (Shustovsky) [inline mw-webapp/src/component/header/Header.tsx:36]: i think use {" "} is not good practice

File: mw-webapp/src/logic/aboutProjectPage/AboutProjectPage.module.scss [hunk 2]
```diff
@@ -8,3 +8,19 @@
...
+  top: 1rem;
+  left: 44%;
+  color: $lightyellow;
+  text-shadow: 2px 2px 2px $button-shadow;
+}
+
+.titleWhenLoggedIn {
+  position: absolute;
+  top: 2.2rem;
+  left: 44%;
```
Comment (Shustovsky) [inline mw-webapp/src/logic/aboutProjectPage/AboutProjectPage.module.scss:23]: i think here need to use variables

File: mw-webapp/src/logic/aboutProjectPage/AboutProjectPage.tsx [hunk 3]
```diff
@@ -49,6 +65,7 @@ export const AboutProjectPage = () => {
       <Title
         level={HeadingLevel.h2}
         text="About project page"
+        className={titleStyles}
```
Comment (Shustovsky) [inline mw-webapp/src/logic/aboutProjectPage/AboutProjectPage.tsx:68]: i think you can use just ternary operator(or clsx) instead of useState+ useEffect

File: mw-webapp/src/logic/allWaysPage/AllWaysPage.module.scss [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  text-shadow: 2px 2px 2px $button-shadow;
+}
+
+.titleWhenLoggedIn {
+  position: absolute;
+  top: 2.2rem;
+  left: 47%;
+  color: $lightyellow;
+  text-shadow: 2px 2px 2px $button-shadow;
+}
```
Comment (Shustovsky) [inline mw-webapp/src/logic/allWaysPage/AllWaysPage.module.scss:17]: DRY. these styles are repeated

File: mw-webapp/src/logic/wayPage/WayPage.tsx [hunk 5]
```diff
@@ -1,22 +1,41 @@
...
 import styles from "src/logic/wayPage/WayPage.module.scss";
 
 /**
  * Way page
  */
 export const WayPage = () => {
   const {uuid} = useParams<QueryParamTypes["uuid"]>();
+  const [user, setUser] = useState<User | null>(null);
+  const [titleStyles, setTitleStyles] = useState(styles.titlePage);
+  // eslint-disable-next-line no-console
```
Comment (Shustovsky) [inline mw-webapp/src/logic/wayPage/WayPage.tsx:19]: please remove it

File: mw-webapp/src/component/header/Header.module.scss [hunk 6]
```diff
@@ -11,7 +11,15 @@
 
 .title {
   display: flex;
-  align-items: center;
+  align-items: flex-start;
+  justify-content: center;
+  color: $button-active;
+  text-shadow: 2px 2px 2px $button-shadow;
+  margin-top: -1em;
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/header/Header.module.scss:18]: Is it right that title hase button styles?

Comment (Ekaterina1994) [inline mw-webapp/src/component/header/Header.module.scss:18]: Why?

File: mw-webapp/src/component/header/Header.module.scss [hunk 7]
```diff
@@ -11,7 +11,15 @@
...
+}
+
+.logo {
+  display: flex;
   justify-content: center;
   color: $button-active;
   text-shadow: 2px 2px 2px $button-shadow;
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/header/Header.module.scss:25]: logo can't have button styles

File: mw-webapp/src/component/header/Header.tsx (hunk 1 above)
Comment (Ekaterina1994) [inline mw-webapp/src/component/header/Header.tsx:36]: Why did you change text?

File: mw-webapp/src/logic/aboutProjectPage/AboutProjectPage.module.scss [hunk 8]
```diff
@@ -8,3 +8,19 @@
...
+  color: $lightyellow;
+  text-shadow: 2px 2px 2px $button-shadow;
+}
```
Comment (Ekaterina1994) [inline mw-webapp/src/logic/aboutProjectPage/AboutProjectPage.module.scss:26]: title is not button

File: mw-webapp/src/logic/aboutProjectPage/AboutProjectPage.module.scss [hunk 9]
```diff
@@ -8,3 +8,19 @@
...
+
+.titlePage {
+  position: absolute;
+  top: 1rem;
+  left: 44%;
+  color: $lightyellow;
+  text-shadow: 2px 2px 2px $button-shadow;
+}
+
+.titleWhenLoggedIn {
```
Comment (Ekaterina1994) [inline mw-webapp/src/logic/aboutProjectPage/AboutProjectPage.module.scss:20]: Are you sure that we need so many title styles? We don't have design yet

File: mw-webapp/src/logic/aboutProjectPage/AboutProjectPage.tsx [hunk 10]
```diff
@@ -1,13 +1,29 @@
...
+
+  useEffect(() => {
+    handleUserAuthState(setUser);
+  }, []);
+
+  useEffect(() => {
+    if (user) {
+      setTitleStyles(styles.titleWhenLoggedIn);
+    }
+
```
Comment (Ekaterina1994) [inline mw-webapp/src/logic/aboutProjectPage/AboutProjectPage.tsx:25]: Why you desided to use useState and useEffect for styles?

File: mw-webapp/src/logic/allWaysPage/AllWaysPage.module.scss (hunk 4 above)
Comment (Ekaterina1994) [inline mw-webapp/src/logic/allWaysPage/AllWaysPage.module.scss:17]: Will you use these styles for title on each of 8 pages?

File: mw-webapp/src/logic/allWaysPage/AllWaysPage.tsx [hunk 11]
```diff
@@ -1,15 +1,34 @@
...
+  useEffect(() => {
+    handleUserAuthState(setUser);
+  }, []);
+
+  useEffect(() => {
+    if (user) {
+      setTitleStyles(styles.titleWhenLoggedIn);
+    }
+
+  }, [user]);
```
Comment (Ekaterina1994) [inline mw-webapp/src/logic/allWaysPage/AllWaysPage.tsx:24]: Why useEffect for titles?

File: mw-webapp/src/logic/wayPage/WayPage.module.scss [hunk 12]
```diff
@@ -1,4 +1,22 @@
...
+  text-shadow: 2px 2px 2px $button-shadow;
+}
+
+.titleWhenLoggedIn {
+  position: absolute;
+  top: 2.2rem;
+  left: 47%;
+  color: $lightyellow;
+  text-shadow: 2px 2px 2px $button-shadow;
+}
```
Comment (Ekaterina1994) [inline mw-webapp/src/logic/wayPage/WayPage.module.scss:22]: Its a bad idea to use so many styles for title on each page

File: mw-webapp/src/logic/wayPage/WayPage.tsx [hunk 13]
```diff
@@ -1,22 +1,41 @@
...
+  useEffect(() => {
+    handleUserAuthState(setUser);
+  }, []);
+
+  useEffect(() => {
+    if (user) {
+      setTitleStyles(styles.titleWhenLoggedIn);
+    }
+
+  }, [user]);
```
Comment (Ekaterina1994) [inline mw-webapp/src/logic/wayPage/WayPage.tsx:31]: Don't use

File: mw-webapp/src/styles/_variables.scss [hunk 14]
```diff
@@ -15,6 +15,7 @@ $progressBar-bg: rgb(255, 255, 255);
 $font-size-large: 48px;
 
 // Colors
+$lightyellow: rgb(246, 246, 150);
 $lightblue: rgb(85, 122, 149);
 $lightgray: rgb(211, 211, 211);
 $lightgray2: #efefef;
```
Comment (Ekaterina1994) [inline mw-webapp/src/styles/_variables.scss:21]: I think that better to use words primary, secondary and etc for colors vars/ Because if you will want to change design in this case you will need to rewrite all colors vars in each files but if you will use primary, secondary and etc you will need to change color only in global vars file

**Review by Ekaterina1994 — CHANGES_REQUESTED** [review]
Please check name of PR

File: mw-webapp/src/component/header/Header.tsx (hunk 1 above)
Comment (MaratReactJs) [reply mw-webapp/src/component/header/Header.tsx]: returned it as it was

File: mw-webapp/src/logic/aboutProjectPage/AboutProjectPage.module.scss (hunk 2 above)
Comment (MaratReactJs) [reply mw-webapp/src/logic/aboutProjectPage/AboutProjectPage.module.scss]: done via mixin

File: mw-webapp/src/logic/aboutProjectPage/AboutProjectPage.tsx (hunk 3 above)
Comment (MaratReactJs) [reply mw-webapp/src/logic/aboutProjectPage/AboutProjectPage.tsx]: thanks via clsx really better

File: mw-webapp/src/logic/allWaysPage/AllWaysPage.module.scss (hunk 4 above)
Comment (MaratReactJs) [reply mw-webapp/src/logic/allWaysPage/AllWaysPage.module.scss:10]: done via mixin

File: mw-webapp/src/logic/wayPage/WayPage.tsx (hunk 5 above)
Comment (MaratReactJs) [reply mw-webapp/src/logic/wayPage/WayPage.tsx]: removed

File: mw-webapp/src/component/header/Header.module.scss (hunk 6 above)
Comment (MaratReactJs) [reply mw-webapp/src/component/header/Header.module.scss:18]: changed

Comment (MaratReactJs) [reply mw-webapp/src/component/header/Header.module.scss:18]: changed

File: mw-webapp/src/component/header/Header.module.scss (hunk 7 above)
Comment (MaratReactJs) [reply mw-webapp/src/component/header/Header.module.scss]: changed

File: mw-webapp/src/component/header/Header.tsx (hunk 1 above)
Comment (MaratReactJs) [reply mw-webapp/src/component/header/Header.tsx]: returned it as it was

File: mw-webapp/src/logic/aboutProjectPage/AboutProjectPage.module.scss (hunk 8 above)
Comment (MaratReactJs) [reply mw-webapp/src/logic/aboutProjectPage/AboutProjectPage.module.scss:19]: changed

File: mw-webapp/src/logic/allWaysPage/AllWaysPage.module.scss (hunk 4 above)
Comment (MaratReactJs) [reply mw-webapp/src/logic/allWaysPage/AllWaysPage.module.scss:10]: I only know about three pages, I made mixins

File: mw-webapp/src/logic/allWaysPage/AllWaysPage.tsx (hunk 11 above)
Comment (MaratReactJs) [reply mw-webapp/src/logic/allWaysPage/AllWaysPage.tsx]: changed to clsx

File: mw-webapp/src/logic/wayPage/WayPage.module.scss (hunk 12 above)
Comment (MaratReactJs) [reply mw-webapp/src/logic/wayPage/WayPage.module.scss:15]: I made mixins

File: mw-webapp/src/logic/wayPage/WayPage.tsx (hunk 13 above)
Comment (MaratReactJs) [reply mw-webapp/src/logic/wayPage/WayPage.tsx]: I made mixins

File: mw-webapp/src/styles/_variables.scss (hunk 14 above)
Comment (MaratReactJs) [reply mw-webapp/src/styles/_variables.scss:21]: We already discussed this in the chat, but in the end we stuck with this option

File: mw-webapp/src/component/header/Header.module.scss (hunk 6 above)
Comment (Ekaterina1994) [reply mw-webapp/src/component/header/Header.module.scss:18]: It's not good idea to use for color rgd(...). Better to use global var with color )

Comment (Ekaterina1994) [reply mw-webapp/src/component/header/Header.module.scss:18]: Why margin-top === -1em?

File: mw-webapp/src/component/header/Header.module.scss [hunk 15]
```diff
@@ -11,10 +11,18 @@
...
+  margin-top: -1em;
+}
+
+.logo {
+  display: flex;
   justify-content: center;
-  color: $button-active;
-  text-shadow: 2px 2px 2px $button-shadow;
+  color: $lightyellow;
+  text-shadow: 2px 2px 2px rgb(77, 77, 77);
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/header/Header.module.scss:25]: Please don't use rgb for color/ Use vars

File: mw-webapp/src/logic/aboutProjectPage/AboutProjectPage.module.scss (hunk 9 above)
Comment (Ekaterina1994) [reply mw-webapp/src/logic/aboutProjectPage/AboutProjectPage.module.scss:17]: Are you sure that  we need different styles for one title (titlePage, titleWhenLoggedIn, etc) It's too amny styles for just one title.

File: mw-webapp/src/logic/aboutProjectPage/AboutProjectPage.tsx [hunk 16]
```diff
@@ -1,13 +1,22 @@
...
 
 /**
  * About project page
  */
 export const AboutProjectPage = () => {
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    handleUserAuthState(setUser);
+  }, []);
```
Comment (Ekaterina1994) [inline mw-webapp/src/logic/aboutProjectPage/AboutProjectPage.tsx:19]: Dont know clear but something looks wrong. It's strange that you using state for userAuth on each page. 
I think better delete title from all pages or move it to header

File: mw-webapp/src/styles/_variables.scss (hunk 14 above)
Comment (Ekaterina1994) [reply mw-webapp/src/styles/_variables.scss:21]: Maybe I something didn't saw. You can ask again this question// Now we have designer or ask Victor...

**Review by Ekaterina1994 — CHANGES_REQUESTED** [review]
Please check name of your PR )
